nservicebus/sqlserver/operations-scripting.md

```diff
@@ -90,3 +90,44 @@ snippet:sqlserver-return-to-source-queue
 ### Using the retry helper methods
 
 snippet:sqlserver-return-to-source-queue-usage
+
+
+### How to shift SqlTransport audit log to long term storage
+
+There are several ways to achieve this. If you have access to a Database Administrator talk to them about techniques like [Table Partitioning](https://technet.microsoft.com/en-us/library/ms188730.aspx) and [Snapshot Replication](https://technet.microsoft.com/en-us/library/ms151832.aspx).
+
+If you're just looking for some simple techniques to clear down your audit queue try this
+
+In SQL Management Studio create an audit_archive table with this SQL script.
+
+```
+CREATE TABLE [dbo].[audit_archive](
+	[Id] [uniqueidentifier] NOT NULL,
+	[CorrelationId] [varchar](255) NULL,
+	[ReplyToAddress] [varchar](255) NULL,
+	[Recoverable] [bit] NOT NULL,
+	[Expires] [datetime] NULL,
+	[Headers] [varchar](max) NOT NULL,
+	[Body] [varbinary](max) NULL,
+	[RowVersion] [bigint] NOT NULL
+) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
+
+```
+
+Next run this script. 
+
+```
+DELETE FROM [dbo].[audit] 
+OUTPUT [deleted].*
+INTO [dbo].[audit_archive] 
+```
+
+You could run this with a scheduled job if you want to clear down the archive regularly. The script moves the contents of the audit table into audit_archive table you created earlier. You also could wrap this in a transaction.
```

Comment:
> Flesh out why a transaction would be needed

I thought about that but I didn't want the information to turn into a tutorial on how to use SQL. I think if people have got their hands grubby at this level there is enough info here to guide them but not so much that we prescribe a solution. Remember I do suggest that they talk to their DBA before doing all this.

> better to do it in batches, not DELETE

The SQL command here is essentially a move command if you wanted to batch by only shifting the oldest 1000 records you could do so by adjusting the DELETE clause but then you would introduce more table scanning etc.

The idea with including this information is to let people know it could be done but not prescribe a full solution. Like you suggest each system is different with different issues around disk management and scale.